lib/ABI/FunctionType/Layout.cpp

```diff
@@ -332,9 +332,9 @@ ToRawConverter::distributePositionBasedArguments(const ArgumentSet &Arguments,
 
   for (const model::Argument &Argument : Arguments) {
     std::size_t RegisterIndex = Argument.Index() + SkippedCount;
-    if (Result.size() <= RegisterIndex)
-      Result.resize(RegisterIndex + 1);
-    auto &Distributed = Result[RegisterIndex];
+    if (Result.size() <= Argument.Index())
+      Result.resize(Argument.Index() + 1);
```

Comment:
I'll look into it.

Reply:
It looks like this was a leftover of some older, more complex design which I was mindlessly carrying over.

So, I just removed it: https://github.com/revng/revng/pull/232/commits/78927e3ac392b74f18bdd5c398161ec4ebc7957e.

I'll keep an eye open for problems with it just in case, but I don't think there are going be any.